### DISCUSSION TITLE
Example of error reporting

### DISCUSSION THREAD

**Discussion by pdumais** [freeconf/restconf#16] [General]
Could we get an example of error reporting with restconf? For example, within onAction, if we'd like to return a 401 with a payload as defined in section 7.1 of rfc8040 ?

**Comment by dhubler** [comment]
I just created this doc, should be what you need.


https://github.com/freeconf/restconf/wiki/Advanced#returning-401-409-and-400-class-errors

**Reply by pdumais** [reply]
Thanks. So would restconf take care of filling put thw payload such as 
`HTTP/1.1 409 Conflict
      Date: Thu, 26 Jan 2017 20:56:30 GMT
      Server: example-server
      Content-Type: application/yang-data+json

      {
        "ietf-restconf:errors" : {
          "error" : [
            {
              "error-type" : "protocol",
              "error-tag" : "lock-denied",
              "error-message" : "Lock failed; lock already held"
            }
          ]
        }
      }`
? Or would we need to set the raw json to do this?

**Comment by dhubler** [comment]
oops, Freeconf/Restconf currently doesn't set that payload beyond plain
text response of the error message text and the HTTP status code.  This
would have to be added, BUT it wouldn't be that hard at all.  Can you
create an issue and pointers to the spec?

**Comment by pdumais** [comment]
Thanks. I created https://github.com/freeconf/restconf/issues/17
